app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -122,6 +125,63 @@ object FirebaseDatabaseUtils {
         )
     }
 
+    /**
+     * Helper function to obtain the posts of a tournament from the database
+     * @param data the data snapshot containing the posts
+     * @return a list containing the posts of the tournament
+     */
+    fun transformPostList(data: DataSnapshot?): List<TournamentPost> {
+        return data?.children?.mapNotNull {
+            transformPost(it)
+        } ?: emptyList()
+    }
+
+    /**
+     * Helper function to obtain a post from a database snapshot
+     * @param data the data snapshot containing the post
+     * @return the post, or null if an error occurred
```

Comment:
Suggestion: Consider using an exception instead of returning null to indicate errors

Reply:
Thanks for the suggestion, however I really don't think throwing an Exception in these cases is a better idea (but in nearly all cases I would completely agree). The reason is that it is much easier to handle "corrupted" data this way by using function like `mapNotNull` to handle lists of Post, Run, ... That way the corrupted data is simply ignored with a message in the logs, and we don't require endless try-catch blocks to prevent the app from crashing.